urnai/actions/chain_of_actions_base.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
+
+from action_base import ActionBase
+
+from urnai.utils.returns import List
+
+
+class ChainOfActionsBase(ABC):
+
+	"""This class works as a sequence of actions"""
+
+	@abstractmethod
+	def get_actions(self) -> List[ActionBase]:
+		"""Returns all the actions in the sequence."""
+		...
+	
+	@abstractmethod
+	def get_first(self) -> ActionBase:
+		"""Returns the first action in the sequence"""
+		...
+	
+	@abstractmethod
+	def add_action(self, action: ActionBase):
+		"""Contains logic for adding an action to the sequence"""
+		...
+		
+	@abstractmethod
+	def remove_action(self, action: ActionBase):
+		"""Contains logic for removing an action from the sequence"""
+		...
+		
+	@abstractmethod
+	def clear():
+		...
+	
+	@abstractmethod
+	def merge(self, chain_of_actions):
+		"""Contains logic for merging a list of sequences of actions into self"""
+		...
+	
+	# Executing the next action in the sequence
+	def next(self):
```

Comment:
I think the function comment should be below and not above the function signature.